libs/langchain/langchain/llms/watsonxllm.py

```diff
@@ -0,0 +1,172 @@
+    """
+    IBM watsonx.ai large language models.
+
+    To use, you should have ``ibm_watson_machine_learning`` python package installed,
+    and the environment variable ``WATSONX_APIKEY`` set with your API key, or pass
+    it as a named parameter to the constructor.
+
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import WatsonxLLM
+            llm = WatsonxLLM(
+                model_id='google/flan-ul2',
+                credentials = {"url": "https://us-south.ml.cloud.ibm.com",
+                                   "apikey": "*****"},
+                project_id="*****"
+            )
+    """
+
+    model_id: str = ""
+    """Type of model to use."""
+
+    project_id: str = ""
+    """ID of the Watson Studio project."""
+
+    space_id: str = ""
+    """ID of the Watson Studio space."""
+
+    credentials: dict = {}
+    """Credentials to Watson Machine Learning instance"""
+
+    params: dict = {}
+    """Model parameters to use during generate requests."""
+
+    verify: Union[str, bool] = ""
+    """User can pass as verify one of following:
+        the path to a CA_BUNDLE file
+        the path of directory with certificates of trusted CAs
+        True - default path to truststore will be taken
+        False - no verification will be made"""
+
+    llm_type: str = "IBM watsonx.ai"
+    """Type of llm."""
+
+    watsonx_model: Any
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that credentials and python package exists in environment."""
+        if "url" not in values["credentials"]:
+            raise TypeError("`url` is not provided.")
+        if "cloud.ibm.com" in values["credentials"].get("url", ""):
+            values["credentials"]["apikey"] = get_from_dict_or_env(
+                values["credentials"], "apikey", "WATSONX_APIKEY"
+            )
+        else:
+            if (
+                "token" not in values["credentials"]
+                and "TOKEN" not in os.environ
+                and "password" not in values["credentials"]
+                and "PASSWORD" not in os.environ
+                and "apikey" not in values["credentials"]
+                and "WATSONX_APIKEY" not in os.environ
+            ):
+                raise ValueError(
+                    "Did not find 'token', 'password' or 'apikey',"
+                    " please add an environment variable"
+                    " `TOKEN`, 'PASSWORD' or 'WATSONX_APIKEY' which contains it,"
+                    " or pass 'token', 'password' or 'apikey'"
+                    " as a named parameter in `credentials`."
+                )
+            elif "token" in values["credentials"] or "TOKEN" in os.environ:
+                values["credentials"]["token"] = get_from_dict_or_env(
+                    values["credentials"], "token", "TOKEN"
```

Comment:
the env variable names should include watsonx in the prefix

Reply:
All env variable have been changed